api/api/src/main/scala/uk/ac/wellcome/platform/api/models/WorkFilter.scala

```diff
@@ -4,7 +4,7 @@ import java.time.LocalDate
 
 sealed trait WorkFilter
 
-case class ItemLocationTypeFilter(locationTypeIds: Seq[String])
+case class ItemsLocationLocationTypeFilter(locationTypeIds: Seq[String])
```

Comment:
:open_mouth: what is the reason for this name change? I'm sure there is a better alternative, but not sure quite sure what you want to signify with it.

Reply:
Ah, it was part of something else, happy to pull it out of this PR - basically to follow suit with all other namings of having it the same as the parameter, but `camelCased`

Reply:
Ahhh right! But aren't the parameter `period.seperated` paths rather than `camelCased`?

Either way, personally I don't reckon it makes the code easier to read having the case class names follow this convention too, and prefer the more descriptive `ItemLocationTypeFilter`. Even better IMO we could do what we do elsewhere and have `WorkFilter.ItemLocationType` etc.

If you think having the classes name reflect the query parameters is an improvement then feel 
 free to go ahead though.

Reply:
The path is `item.locations.locationType` <- I am just replicating this to leave no room for guessing if it's referencing the same thing... personally I prefer consistency over making it slightly easier to read, as easy to read doesn't always help with new people coming to the code base.

I'll remove it from the PR, and do it universally.

Reply:
Well I meant the code was easier to read in the broader sense than just reading out the name (maybe should have said easier to understand). It is not clear when encountering a filter in the code that it is named in such a way that represents a path, and also means more detail is encoded in the name than it need to (we can't move things around in the model without  remembering to change the name of the class, introducing unnecessary coupling etc)

Reply:
For what is worth I agree with @warrd on this. Especially on the coupling part of it. There is no way that we can guarantee that the name will always match the path in the JSON because there are definitely going to be instances where we will make changes in the model and we will forget to change the class. 
I think making a claim that this is always going to match the json path could make things more difficult for people newer to the codebase, rather than easier.